faunadb/client.py

```diff
@@ -404,3 +411,12 @@ def _perform_request(self, action, path, data, query, headers):
         req = Request(action, url, params=query, data=to_json(
             data), auth=self.auth, headers=headers)
         return self.session.send(self.session.prepare_request(req))
+
+    def _get_tags_string(self, tags_dict):
+        if not isinstance(tags_dict, dict):
+            raise Exception("Tags must be a dictionary")
+        
+        return ",".join(["=".join([k, tags_dict[k]]) for k in tags_dict])
+
+    def _is_valid_traceparent(self, traceparent):
+        return bool(re.match("^[\da-f]{2}-[\da-f]{32}-[\da-f]{16}-[\da-f]{2}$", traceparent))
```

Comment:
Does our downstream care if this is a valid traceparent? i.e. does it do something with the individual components and therefore expect it to be valid, or do we not really care? If not, we may wish to be extremely permissive here (less code to maintain if something changes).

If we really do care, note that version `ff` is invalid per the spec, and all zeroes are invalid for trace-id and parent-id. Although I don't know if that matters, sooo.. :D

Reply:
This regex is lifted directly from the JS driver - I think we're mainly going for consistency in behavior.